python/mxnet/test_utils.py

```diff
@@ -34,6 +34,7 @@
 import numpy as np
 import numpy.testing as npt
 import numpy.random as rnd
+import scipy.stats as ss
```

Comment:
we don't depend on scipy

Reply:
How should I revise this? Move it to be inside the functions?

Reply:
see mx.nd.sparse

Reply:
@javelinjs @piiswrong I've added one commit to solve the problem. Also, 
 I also add tests for the case in which the generator is triggered multiple times. https://github.com/apache/incubator-mxnet/pull/9129/commits/199fabdcc06736393f63639a806f522454341d02